vispy/visuals/volume.py

```diff
@@ -723,15 +715,16 @@ def _build_clipping_planes_func(self):
         """
         build the code snippet used to clip the volume based on self.clipping_planes
         """
-        func = Function('$vars\nfloat clip_planes(vec3 loc) { float is_shown = 1.0; $clips; return is_shown; }')
+        func = Function(
+            '$vars\nfloat clip_planes(vec3 loc, vec3 vol_shape) { float is_shown = 1.0; $clips; return is_shown; }')
```

Comment:
Actually, this does not strictly need to be a function, and since it's being used only once maybe it's simply better to leave everything on the outer scope (like `$vars` at the beginning). this may even have a performance benefit.

In either case, you actually don't need to pass the volume shape as a parameter, cause the function can use the u_shape from the global scope (afaik).

Reply:
I tried using `u_shape` from the global scope and it complained about undefined usage (I don't get why, because you're already using other uniforms in the function)

I assume that the functions get inlined on shader compilation but not 100% of that so there's a possible performance benefit yeah!

Reply:
I can't figure out why this doesn't work. For now, I'll merge as is!